### PR TITLE
code climate integration

### PR DESCRIPTION
**Pull request by adireddy**
- code climate integration (coming soon)

### CODE REVIEW

File: src/checkstyle/checks/Check.hx [hunk 1]
```diff
@@ -12,17 +12,25 @@ class Check {
...
+	public var categories:Array<String>;
+	public var points:Int;
+	public var desc:String;
 
-	var messages:Array<LintMessage>;
+	var messages:Array<CheckMessage>;
 	var moduleName:String;
 	var checker:Checker;
 
+	// Categories: Bug Risk, Clarity, Compatibility, Complexity, Duplication, Performance, Security, Style
```
Comment (Gama11) [inline src/checkstyle/checks/Check.hx:23]: Seems like there should be a category (abstract) enum?

Comment (adireddy) [inline src/checkstyle/checks/Check.hx:23]: The above are the only categories supported by code climate, generic to all languages.

Comment (Gama11) [inline src/checkstyle/checks/Check.hx:23]: I didn't mean adding a category named "enum", but creating an enum for the categories so you don't have string literals everywhere. :)

Comment (adireddy) [inline src/checkstyle/checks/Check.hx:23]: Sure :) makes more sense now.

**Comment by adireddy** [comment]
I have added `categories` and `points` for each check. Let me know if you want to change any categories.

The points I have assigned are Fibonacci numbers based on the complexity to fix the issue.

Minimum is 1 (for example to remove an extra empty line, space, unused import, etc)
1, 2, 3, 5, 8, 13, 21, 34, etc

Let me know if you want to edit these numbers as well.

@AlexHaxe @Gama11

File: src/checkstyle/Checker.hx [hunk 2]
```diff
@@ -134,10 +137,15 @@ class Checker {
 			var flagValue = define.split("=");
 			parser.define(flagValue[0], flagValue[1]);
 		}
-		return parser.parse();
+
+		try {
+			return parser.parse();
+		}
+		catch (e:Dynamic) {}
```
Comment (Gama11) [inline src/checkstyle/Checker.hx:144]: I don't like this, this creates another "swallowed errors" problem (see #154). It makes sense to continue, but we should still output the error.

**Comment by Gama11** [comment]
Shouldn't all checks have some categories and a points value?

File: src/checkstyle/reporter/CodeClimateReporter.hx [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package checkstyle.reporter;
+
+import checkstyle.CheckMessage.SeverityLevel;
+import haxe.Json;
+
+class CodeClimateReporter extends BaseReporter {
+
+	static var INFO:String = "info";
```
Comment (Gama11) [inline src/checkstyle/reporter/CodeClimateReporter.hx:8]: Seems like these are only used in `getSeverity()`, so does it make sense to have them in static vars?

File: test/misc/ExtensionsTest.hx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package misc;
+
+import checks.CheckTestCase;
+import checkstyle.checks.whitespace.IndentationCharacterCheck;
+
+class ExtensionsTest extends CheckTestCase<ExtensionsTests> {
```
Comment (Gama11) [inline test/misc/ExtensionsTest.hx:6]: Why `ExtensionsTest`? Doesn't this belong in `IndentationCharacterCheck`?

**Comment by adireddy** [comment]
By default, all the checks have categorized as `Style` with `1` point and can be overridden in each check.

File: test/misc/ExtensionsTest.hx (hunk 4 above)
Comment (adireddy) [inline test/misc/ExtensionsTest.hx:6]: Just added to reproduce haxeparser exception and masked it with try/catch.

File: src/checkstyle/reporter/CodeClimateReporter.hx (hunk 3 above)
Comment (adireddy) [inline src/checkstyle/reporter/CodeClimateReporter.hx:8]: they are supposed to be `inlines`. Changed them.

File: test/misc/ExtensionsTest.hx (hunk 4 above)
Comment (adireddy) [inline test/misc/ExtensionsTest.hx:6]: we can use `misc` to add generic tests that don't belong to any check.

**Comment by Gama11** [comment]
Ah, okay. In that case it seems redundant to have `points = 1;` assignemnts in child classes though?

**Comment by adireddy** [comment]
yes `points = 1;` can be removed

**Pull request merged**